protostar/cheatable_starknet/cheaters/expects.py

```diff
@@ -0,0 +1,112 @@
+import os
+from typing import Optional, Callable, TYPE_CHECKING
+
+from starkware.starknet.public.abi import get_selector_from_name
+
+from protostar.starknet import CheatcodeException
+from protostar.starknet.cheater import CheaterException
+from protostar.starknet.address import Address
+from protostar.starknet.data_transformer import (
+    CairoOrPythonData,
+    transform_calldata_to_cairo_data,
+    CairoData,
+)
+from protostar.starknet.types import SelectorType
+
+
+if TYPE_CHECKING:
+    from protostar.cheatable_starknet.cheatable_cached_state import CheatableCachedState
+
+
+class ExpectsCheaterException(CheaterException):
+    pass
+
+
+class ExpectsCairoCheater:
+    def __init__(self, cheatable_state: "CheatableCachedState"):
+        self.cheatable_state = cheatable_state
+
+    async def expect_call(
+        self,
+        contract_address: Address,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> Callable:
+        contract_address_int = int(contract_address)
+        cairo_calldata = await transform_calldata_to_cairo_data(
+            contract_class=await self.cheatable_state.get_contract_class(
+                await self.cheatable_state.get_class_hash_at(
+                    contract_address=contract_address_int
+                )
+            ),
+            function_name=function_name,
+            calldata=calldata,
+        )
+
+        self.register_expected_call(
+            contract_address=contract_address,
+            function_selector=get_selector_from_name(function_name),
+            calldata=cairo_calldata,
+        )
+
+        def stop_callback():
+            try:
+                ExpectsCairoCheater.assert_no_expected_calls(
+                    expected_calls=self.cheatable_state.expected_contract_calls,
+                    fixed_address=contract_address,
+                )
+            except ExpectsCheaterException as e:
+                raise CheatcodeException("expect_call", e.message) from e
```

Comment:
Is this ok? `stop_callback` is going to be called from hints:
```
stop_expect = expect_call(addr, "increase_balance", [50])
stop_expect()
```

Reply:
I'm not completely sure if we still will be able to return callbacks in cairo 1.0. Probably we'll use an id-based solution (like `setTimeout`/`clearTimeout` does). Base class of the cheatcode states that it will be wrapped in a `ExecutionResult` anyways right now